File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    final double producedPerHour = 221;

    double getSuccessRateBySpeed(int speed) {
        if (1 <= speed && speed <= 4) {
            return 1.0;
        } else if (5 <= speed && speed <= 8) {
            return 0.9;
        } else if (speed == 9) {
            return 0.8;
        } else {
            return 0.77;
        }
    }

    public double productionRatePerHour(int speed) {
        double successRate = this.getSuccessRateBySpeed(speed);
        double carsPerHour = this.producedPerHour * speed;

        return carsPerHour * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        int carsPerHour = (int) this.productionRatePerHour(speed);

        return carsPerHour / 60;
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{

    private const int ProductionRate = 221;
    
    public static double SuccessRate(int speed)
    {
        return speed switch
        {
            <= 4 and >= 1 => 1.0,
            <= 8 and >= 5 => 0.9,
            9 => 0.8,
            10 => 0.77,
            _ => 0.0
        };
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        var perHour = ProductionRate * speed;
        var successRate = SuccessRate(speed);
        return successRate * perHour;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        var ratePerHour = ProductionRatePerHour(speed);

        return (int)ratePerHour / 60;
    }
}
